pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -336,6 +336,16 @@ class Timestamp(_Timestamp):
         """
         return cls(datetime.combine(date, time))
 
+    @classmethod
+    def fromisocalendar(cls, year, week, day):
+        """
+        Timestamp.fromisocalendar(year, week, day)
+
+        Return a new Timestamp corresponding to the
+        ISO calendar date specified by year, week and day.
+        """
+        return cls(datetime.fromisocalendar(year, week, day))
```

Comment:
You've got the right idea, but keep in mind that fromisocalendar doesn't exist in py36 or py37, so you'll need to re-do some of the logic here

Alternatively, you could just make this raise NotImplementedError for earlier versions

Reply:
Is that better?